Sourcery/Models/TypeName.swift

```diff
@@ -26,27 +26,35 @@ protocol Typed {
 extension Typed {
     var isOptional: Bool { return typeName.isOptional }
     var unwrappedTypeName: String { return typeName.unwrappedTypeName }
+    var actualTypeName: TypeName? { return typeName.actualTypeName }
+    var isTuple: Bool { return typeName.isTuple }
 }
 
 // sourcery: skipDescription
 final class TypeName: NSObject, AutoDiffable {
     let name: String
 
+    /// Actual type name if given type name is type alias
```

Comment:
Before we had only `typeName` and `type`, and type will be set as actual type even when type is an alias, but only if it is an alias for known type. It will not work in case of types that are not `Type` at all, like closures and tuples. So when resolving type we now not just set the `type` property, but also the name of the actual type. Nothing else changes.
I think it makes sense to allow access to both type names so user can decide what to use - what is used in code, or what will be used in runtime.

Also in future we should improve how we handle typealiases for optional types, i.e. `typealias Handler = (()->())?`. Not saying that this is a "good" practice, but it's just valid Swift, so maybe we should support that too, or just agree to ignore that case for now.